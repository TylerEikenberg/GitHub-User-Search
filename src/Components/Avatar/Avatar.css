.Avatar-image-styles {
  border: 2px solid var(--light-font-color);
  border-radius: 50%;
  max-width: 100%;
  max-height: 100%;
  height: 7rem;
  position: relative;
}
