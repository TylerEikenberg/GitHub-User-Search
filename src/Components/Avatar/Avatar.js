import React from "react";
import "./Avatar.css";

function Avatar({ image }) {
  return <img className="Avatar-image-styles" src={image} alt="user" />;
}

export default Avatar;
