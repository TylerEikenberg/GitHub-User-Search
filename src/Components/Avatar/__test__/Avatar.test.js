import React from "react";
import ReactDOM from "react-dom";
import Avatar from "../Avatar";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/";
import TestRenderer from "react-test-renderer";

import tyleravi from "../../../Assets";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Avatar></Avatar>, div);
});
it("Avatar is visible", () => {
  const { getByTestId } = render(<Avatar image={tyleravi}></Avatar>);
  expect(getByTestId("Avatar")).toBeVisible();
});
it("Avatar has className", () => {
  const { getByTestId } = render(<Avatar image={tyleravi}></Avatar>);
  expect(getByTestId("Avatar")).toHaveClass("Avatar-image-styles");
});
it("matches snapshot, without image src", () => {
  const tree = TestRenderer.create(<Avatar></Avatar>).toJSON();
  expect(tree).toMatchSnapshot();
});
it("matches snapshot, with image src", () => {
  const tree = TestRenderer.create(<Avatar image={tyleravi}></Avatar>).toJSON();
  expect(tree).toMatchSnapshot();
});
