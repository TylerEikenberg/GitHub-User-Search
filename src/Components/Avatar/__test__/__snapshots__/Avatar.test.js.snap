// Jest Snapshot v1, https://goo.gl/fbAQLP



exports[`matches snapshot, with image src 1`] = `
<img
  alt="user"
  className="Avatar-image-styles"
  data-testid="Avatar"
/>
`;

exports[`matches snapshot, without image src 1`] = `
<img
  alt="user"
  className="Avatar-image-styles"
  data-testid="Avatar"
/>
`;
